### DISCUSSION TITLE
on NMODL

### DISCUSSION THREAD

**Discussion by ramcdougal** [ctrl-z-9000-times/NEUWON#4] [General]
In the readme, there's a section with TODO notes for NMODL.

If it helps:

NMODL is an extension of the model description language developed for SCoP; published in

Kohn, M. C., Hines, M. L., Kootsey, J. M., & Feezor, M. D. (1994). A block organized model builder. _Mathematical and computer modelling_, 19(6-8), 75-97. https://doi.org/10.1016/0895-7177(94)90190-2

The extensions needed for NEURON that turned mod files into NMODL are discussed in:

Hines, M. L., & Carnevale, N. T. (2000). Expanding NEURON's repertoire of mechanisms with NMODL. _Neural computation_, 12(5), 995-1007 https://doi.org/10.1162/089976600300015475

And in expanded detail with many examples in Chapter 9 of

Carnevale, N. T., & Hines, M. L. (2006). _The NEURON book_. Cambridge University Press.

NEURON's documentation on NMODL is at
https://nrn.readthedocs.io/en/latest/python/modelspec/programmatic/mechanisms/nmodl.html

You might also want to mention that NMODL files are readily available for common ion channels from ModelDB, ChannelPedia, or by automatic conversion from NeuroML.

**Comment by D-McDougall** [comment]
Thanks! I will add this info to the README.
